### PR TITLE
Add validations

### PR DESCRIPTION
**Pull request by Bludidi**
***Hello Reviewer!***

### In this milestone, I have implemented the following:

- [ ]  Add validations for the user model
- [ ] Add validations for the post model
- [ ] title must not exceed 250 chars
- [ ] Create unit tests

### CODE REVIEW

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @Bludidi, 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 

## Required Changes ♻️

- [ ] This part of the project requirement is missing in your project. Kindly work on it 👍🏽. I can only see the request tests in your project but not the model test which is required for this milestone.

image


_Check the comments under the review._


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  it 'The comments_counter should be and integer not less than zero' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'The likes_counter should be an integer not less than zero' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (mahbubul14) [inline spec/models/post_spec.rb]: - [ ] You have added tests for the models but they all should be passed. Kindly work on it and make pass all the tests. 
If the tests pass on your side, you can add a screenshot and ask for the next review. :+1: 

failed_example

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @Bludidi,

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/models/post_spec.rb (hunk 1 above)
Comment (Bludidi) [reply spec/models/post_spec.rb]: ***Hello @mahbubul14***

The tests are passing on my side and I have attached the screenshot 👇👇

Test screenshot

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  it 'The title should be present' do
+    subject.title.nil?
+    expect(subject).to_not be_valid
+  end
+
+  it 'The title should not exceed 250 characters' do
+    subject.title = 'This is my title' * 30
+    expect(subject).to_not be_valid
+  end
```
Comment (Davidosky007) [inline spec/models/post_spec.rb:16]: * As mentioned by the previous CR it is a requirement to Add unit specs for all of your models' methods and validations.
* Currently you are missing tests for `like` and `comments` , Kindly add them 🙏

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Bludidi 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well written PR 👍 
* Good looking readme 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 2 above)
Comment (Bludidi) [reply spec/models/post_spec.rb:16]: ***Hello!***
I have added tests for like and comments models as the previous reviewer suggested.

**Review by Reem-lab — COMMENTED** [review]
Hi @Bludidi, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that have a live demo section in your README file and to have a professional README file you should remove this section till you have your app deployed so you can just add the live demo section in case you have deployed link for your project is present. .✅❗💯🤩✨✨ I noticed you wrote tha way to open it locally but that's not the  place of how to open it locally this is for a deployed project link online💯🚀🚀

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Bludidi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to make the requested changes.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**